pkg/logql/matchers_test.go

```diff
@@ -32,7 +32,7 @@ func Test_match(t *testing.T) {
 				{mustMatcher(labels.MatchEqual, "a", "1")},
 				{
 					mustMatcher(labels.MatchEqual, "b", "2"),
-					mustMatcher(labels.MatchEqual, "c", "3"),
+					mustMatcher(labels.MatchRegexp, "c", "3"),
```

Comment:
Curious about how this was passing before 🤔 Is it because the matcher was being rewritten to `c="3"`?

Reply:
It is. When the value for a label is `=~<literal>`, that's equivalent to `=<literal>` because matchers anchor the regex to the beginning and end of the line with `^()$`